src/pipdeptree/_models/package.py

```diff
@@ -3,37 +3,39 @@
 import re
 from abc import ABC, abstractmethod
 from importlib import import_module
-from importlib.metadata import PackageNotFoundError, metadata, version
+from importlib.metadata import Distribution, PackageNotFoundError, metadata, version
 from inspect import ismodule
-from typing import TYPE_CHECKING
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
 
-from pip._vendor.pkg_resources import Requirement  # noqa: PLC2701
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+from pip._internal.models.direct_url import DirectUrl  # noqa: PLC2701
+from pip._internal.utils.egg_link import egg_link_path_from_sys_path  # noqa: PLC2701
 
 if TYPE_CHECKING:
-    from pip._internal.metadata import BaseDistribution
-    from pip._vendor.pkg_resources import DistInfoDistribution
+    from importlib.metadata import Distribution
 
 
 def pep503_normalize(name: str) -> str:
     return re.sub("[-_.]+", "-", name)
 
 
+def contains_extra(marker: str) -> bool:
+    return re.search(r"\bextra\s*==", marker)
+
+
 class Package(ABC):
     """Abstract class for wrappers around objects that pip returns."""
 
     UNKNOWN_LICENSE_STR = "(Unknown license)"
 
-    def __init__(self, obj: DistInfoDistribution) -> None:
-        self._obj: DistInfoDistribution = obj
-        self.key = pep503_normalize(obj.key)
-
-    @property
-    def project_name(self) -> str:
-        return self._obj.project_name  # type: ignore[no-any-return]
+    def __init__(self, _project_name: str) -> None:
+        self.project_name = pep503_normalize(_project_name)
```

Comment:
Why did you remove `self.key`? It doesn't make since to normalize the package name (users will see `example_py` instead of `Example-Py` which would be confusing for them. We should keep both `self.key` and `self.project_name`.

Reply:
Sorry, maybe I misunderstand what you meant [here](https://github.com/tox-dev/pipdeptree/pull/333#discussion_r1536871750).